source/agora/consensus/validation/Block.d

```diff
@@ -112,6 +115,151 @@ public string isInvalidReason (const ref Block block, in ulong prev_height,
+        - All the enrollments pass validation, which implies:
+            - The Enrolments should be arranged in ascending order by utxo_key
+            - Signature are authentic
+            - The enrollments refer to freeze tx's in this block
+            - The freeze tx's spend an output which must be found with
+              the findUTXO() delegate
+            - Freeze transactions referenced in registration should not be
+              double-spend
+
+    Params:
+        block = The genesisBlock to check
+
+    Returns:
+        `null` if the genesisBlock is valid, a string explaining the reason it
+        is invalid otherwise.
+
+*******************************************************************************/
+
+private string isValidGenesisBlock (const ref Block genesisBlock) nothrow @safe
+{
+    if (genesisBlock == Block.init)
+        return "GenesisBlock: The block is empty";
+
+    if (genesisBlock.header.height != 0)
+        return "GenesisBlock: The height of the block is not 0";
+
+    if (genesisBlock.header.prev_block != Hash.init)
+        return "GenesisBlock: Header.prev_block is not empty";
+
+    if (genesisBlock.txs.length == 0)
+        return "GenesisBlock: Transaction(s) is empty";
+
+    if (!genesisBlock.txs.isSorted())
+        return "GenesisBlock: Transactions are not sorted";
+
+    if (genesisBlock.txs.length >= Block.TxsInBlock)
+        return "GenesisBlock: The number of transactions is out of bounds";
+
+    UTXOSetValue[Hash] utxo_set;
+    foreach (const ref tx; genesisBlock.txs)
+    {
+        if (!(tx.type == TxType.Payment || tx.type == TxType.Freeze))
+            return "GenesisBlock: Transaction type allows only "
+                ~ "Payment and Freeze";
+
+        if (tx.inputs.length != 1)
+            return "GenesisBlock: Transaction must have only one input";
+
+        if (tx.outputs.length == 0)
+            return "GenesisBlock: No output(s) in the transaction";
+
+        Hash tx_hash = tx.hashFull();
+        foreach (idx, const ref output; tx.outputs)
+        {
+            if (output.address == PublicKey.init)
+                return "GenesisBlock: The address of the output(s) is empty"
+                    ~ "in the transaction";
+
+            // disallow negative amounts
+            if (!output.value.isValid())
+                return "GenesisBlock: Output(s) overflow or underflow"
+                    ~ "in the transaction";
+
+            // disallow 0 amount
+            if (output.value == Amount(0))
+                return "GenesisBlock: Value of output is 0"
+                    ~ "in the transaction";
+
+            if (tx.inputs[0].signature != Signature.init
+                && output.address.verify(tx.inputs[0].signature, tx_hash[]))
+                return "GenesisBlock: Input has invalid signature "
+                    ~ "in the transaction";
+
+            const utxo_value = UTXOSetValue(genesisBlock.header.height,
+                tx.type, output);
```

Comment:
```suggestion
            const utxo_value = UTXOSetValue(/* height: */ 0, tx.type, output);
```

In general your approach is good, but if it forces you to break down your statement on multiple lines, look for alternatives.